### ISSUE TITLE
Use pyprojecttoml.apply_configuration() in tests to verify generated pyproject.toml

### ISSUE THREAD

**Issue by diazona** [diazona/setuptools-pyproject-migration#56]
setuptools includes the function `setuptools.config.pyprojecttoml.apply_configuration()`, which takes a `Distribution` object and a `pyproject.toml` file path and applies the settings in the given `pyproject.toml` file to the `Distribution`, overwriting existing fields where they exist. We could use this to help check that the `pyproject.toml` files generated by our package will be parsed by setuptools in the same way as the original `setup.cfg` or `setup.py`.

This is just an idea; we can decide whether to actually do it later.

**Comment by sjlongland** [comment]
Indeed, probably a good idea to apply any sanity checks we can to the output to validate its correctness where possible.

Perhaps even doing so at run-time… so we generate a config file, then validate it (with an option to disable validation).

**Comment by RonnyPfannschmidt** [comment]
i would strongly recommend that, i jsut tried the tool on pytest and it created a invalid config that needed a good amount of changes and fixes

**Comment by diazona** [comment]
Yeah I'm not surprised - this tool isn't ready yet to handle something as complex as pytest. (In fact, you may have noticed, we have a test case that specifically checks its performance on some recent version of pytest.)

We'll probably implement this soon, but I think its impact is going to be somewhat limited because we already have enough checking to know that the tool doesn't properly work on complex `setup.py` files like in pytest. We still have to actually fix the problems. That will probably be a target of the 0.3 release.
